docs/src/dataio_3_migration.rst

```diff
@@ -0,0 +1,293 @@
+fmu-dataio 3.0 migration guide
+================================
+
+This document contains a set of instructions on how to update your code to work
+with ``fmu-dataio`` 3.0. Almost all changes that have taken place are related to 
+fmu-dataio's ``ExportData`` class.
+
+
+ExportData
+-----------------------
+Changes to input arguments  
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The following arguments are deprecated, or have specific input types/values that are deprecated, 
+but with replacements inplace.
+
+ - ``access_ssdl`` is deprecated and replaced by the ``classification`` and ``rep_include`` arguments.
+ - ``classification='asset'`` is deprecated  use ``classification='restricted'`` instead
+ - ``fmu_context='preprocessed'`` is deprecated use argument ``preprocessed=True`` instead.
+ - ``vertical_domain`` now only supports string input with value either ``time`` / ``depth``. Using 
+   a dictionary form to provide a reference together with the ``vertical_domain`` is deprecated, use 
+   the ``domain_reference`` argument instead.
+ - ``workflow`` now only supports string input, example ``workflow='Structural modelling'``.
+ - ``content={'seismic': {'offset': 10}}`` use the key ``stacking_offset`` instead of ``offset``.
+
+
+Following are an example demonstrating several deprecated patterns:
+
+.. code-block:: python
+
+    from fmu.dataio import ExportData
+
+    ExportData(
+      fmu_context='preprocessed', # ⛔️ 
+      access_ssdl={'access_level': 'asset', 'rep_include': True}, # ⛔️ 
+      vertical_domain={'depth': 'msl'}, # ⛔️ 
+      workflow={'reference': 'Structural modelling'}, # ⛔️ 
+      )
+
+Change to this instead 👇:
+
+.. code-block:: python
+
+    from fmu.dataio import ExportData
+
+    ExportData(
+      preprocessed=True, # ✅
+      classification='restricted', # ✅ note the use of 'restricted' instead of 'asset'
+      rep_include=True, # ✅
+      vertical_domain='depth', # ✅
+      domain_reference='msl', # ✅
+      workflow='Structural modelling', # ✅
+      )
```

Comment:
```suggestion
      workflow='Structural modelling', # ✅
    )
```
Alignment here too, if you feel it's an improvement. I think there are some others as well